### DISCUSSION TITLE
Proper js/ts intendation

### DISCUSSION THREAD

**Discussion by kryptamine** [folke/snacks.nvim#269] [Q&A]
Hi everyone! 

I'm trying to configure snacks indent plugin to replace mini.indentscope. 
<img width="983" alt="chat_manager_ts____Work_chatter_apps_backend_src__-_NVIM-2" src="https://github.com/user-attachments/assets/843f7e3e-db77-43e1-844f-d50fde751417" />

I want to hide outer scope indentation lines, only displaying the indentation for the block I'm currently in.
<img width="1137" alt="chat_manager_ts____Work_chatter_apps_backend_src__-_NVIM" src="https://github.com/user-attachments/assets/953c39af-304e-4f4f-ac03-4123a6be08f8" />
Comparing to the snacks, which highlights the outer block indentation, instead of the current block.

So, to sum up, I want to achieve the behaviour from the first screenshot using snacks.indent.

Could you guide me on how to achieve this behavior?

Thank you in advance!

**Comment by dpetka2001** [comment]
It's mentioned in the [docs](https://github.com/folke/snacks.nvim/blob/main/docs/indent.md#%EF%B8%8F-config). You need to set `only_scope = true` I believe. I haven't done this, but this is what the docs say.

**Comment by kryptamine** [comment]
I've tried `only_scope = true` and `only_current = true`, it didn't help

**Reply by dpetka2001** [reply]
I think you also need to disable `indent.scope.treesitter.enabled = false` if i'm not mistaken.

**Reply by kryptamine** [reply]
It didn't help:( Here is my config:
```
return {
  {
    'folke/snacks.nvim',
    priority = 1000,
    lazy = false,
    opts = {
      statuscolumn = { enabled = false },
      quickfile = { enabled = false },
      rename = { enabled = false },
      bufdelete = { enabled = false },
      notifier = { enabled = true },
      terminal = {
        enabled = true,
        win = {
          position = 'float',
          border = 'single',
        },
      },
      words = {
        enabled = true,
        debounce = 200,
        notify_jump = false,
        notify_end = true,
        foldopen = true,
        jumplist = true,
        modes = { 'n' },
      },
      styles = {
        notification = {
          border = 'single',
        },
      },
      indent = {
        enabled = true,
        only_scope = true,
        only_current = true,
        scope = {
          treesitter = {
            enable = false,
          },
        },
      },
      scope = {
        treesitter = {
          enable = false,
        },
      },
    },
    keys = {
      {
        '<leader>lg',
        function()
          Snacks.lazygit()
        end,
        desc = 'Lazygit',
      },
      {
        '<leader>gB',
        function()
          Snacks.gitbrowse()
        end,
        desc = 'Git Browse',
      },
      {
        '<leader>gf',
        function()
          Snacks.lazygit.log_file()
        end,
        desc = 'Lazygit Current File History',
      },
      {
        '<leader>gl',
        function()
          Snacks.lazygit.log()
        end,
        desc = 'Lazygit Log (cwd)',
      },
      {
        '<c-t>',
        function()
          Snacks.terminal.toggle()
        end,
        desc = 'Toggle Terminal',
      },
      {
        ']]',
        function()
          Snacks.words.jump(vim.v.count1)
        end,
        desc = 'Next Reference',
      },
      {
        '[[',
        function()
          Snacks.words.jump(-vim.v.count1)
        end,
        desc = 'Prev Reference',
      },
    },
  },
}
```

**Reply by dpetka2001** [reply]
No it has to be like
```lua
{
  "folke/snacks.nvim",
  opts = {
    indent = {
      indent = {
        only_scope = true,
      },
    },
  },
},
```
But now that I've tried it it shows the indent for the current scope, but also shows the indents for the inner scopes. It just doesn't show the outer scope indents. 

But maybe this is intended because in the docs it says
> only_scope = false, -- only show indent guides of the scope

It says `indent guides` (plural), so maybe that means it shows the guides of the current scope and its inner scopes. Not really sure myself. 
@folke Is this intended behavior for `only_scope = true`?
